src/domains/chart/components/lib-charts/dygraph-chart.tsx

```diff
@@ -394,8 +417,9 @@ export const DygraphChart = ({
             const after = Math.round(xRange[0])
             const before = Math.round(xRange[1])
 
-            if (before <= (chartData.last_entry * 1000)
-              && after >= (chartData.first_entry * 1000)
+            const { first_entry: firstEntry, last_entry: lastEntry } = propsRef.current.chartData
+            if (before <= (lastEntry * 1000)
+              && after >= (firstEntry * 1000)
```

Comment:
could we use predicate named function for this if?

Reply:
good idea, done https://github.com/netdata/dashboard/pull/11/commits/57e48c74ba6d5eed733f0a37a6616f310cb53160